src/test/java/TestLIC7.java

```diff
@@ -55,6 +55,7 @@ public void testNegative() {
     Assert.assertFalse(res);
   }
 
+  /** Test with a Point array of length 2. Should be false but not cause any exception */
   @Test
   public void testShortList() {
     Point[] pts = new Point[2];
```

Comment:
Shouldn't these points be initialised? For example like `{new Point(0,0), new Point(1,1)};`?